applications/GeoMechanicsApplication/custom_utilities/parameters_utilities.cpp

```diff
@@ -12,15 +12,25 @@
 //                   Gennady Markelov
 //
 
-#include "custom_utilities/parameters_utilities.h" 
+#include "custom_utilities/parameters_utilities.h"
 
 namespace Kratos {
 
 Parameters ParametersUtilities::ExtractParameters(const Parameters& rSourceParameters,
                                                         const std::vector<std::string>& rNamesOfParametersToCopy)
 {
+    // The reason we cannot use Parameters.CopyValuesFromExistingParameters(...) function here is that
+    // it throws an exception if the parameter is not found in the source parameters.
     auto result = Parameters{};
-    result.CopyValuesFromExistingParameters(rSourceParameters, rNamesOfParametersToCopy);
+
+    for (const std::string& entry : rNamesOfParametersToCopy)
+    {
+        if (rSourceParameters.Has(entry))
+        {
+            result.AddValue(entry, rSourceParameters[entry]);
+        }
+    }
+
```

Comment:
Now that we have changed the behavior of this function, I'm pondering whether we should rename it to indicate that it will only copy parameters that exist in `rSourceParameters`. The supplied parameter names don't have to be present there and they are silently skipped if they don't. The only suggestion that comes to my mind now is `CopyOptionalParameters`. What do you think?

Reply:
Good suggestion, I brought back the old function so we can use both